src/goto-bmc/goto_bmc_parse_options.cpp

```diff
@@ -0,0 +1,101 @@
+// Author: Fotis Koutoulakis for Diffblue Ltd.
+
+#include "goto_bmc_parse_options.h"
+
+#include <util/exit_codes.h>
+#include <util/message.h>
+
+#include <goto-programs/read_goto_binary.h>
+
+#include <libcprover-cpp/api_options.h>
+#include <libcprover-cpp/verification_result.h>
+
+#include "api.h"
+
+
+goto_bmc_parse_optionst::goto_bmc_parse_optionst(int argc, const char **argv)
+  : parse_options_baset(
+      GOTO_BMC_OPTIONS,
+      argc,
+      argv,
+      std::string("goto-bmc") + CBMC_VERSION)
+{
+}
+
+void print_messages_to_stdout(
+  const api_messaget &message,
+  api_call_back_contextt)
+{
+  // TODO: this is just a hack while a better way to connect the
+  //  parse_options_baset::log to the api callback is found.
+  //  This has to be removed before merging the PR.
+  ui_message_handlert ui_message_handler(cmdlinet{}, "goto-bmc");
+  messaget log{ui_message_handler};
```

Comment:
This has to be fixed properly.
However the older way to create a log by just calling `messaget log;` was not working.

Reply:
Fixed